### DISCUSSION TITLE
Support disabling ISR for the entire application

### DISCUSSION THREAD

**Discussion by brandonchinn178** [vercel/next.js#42207] [Ideas]
Moved from discussion: https://github.com/vercel/next.js/discussions/23917

### Describe the feature you'd like to request

I would like to disable ISR in a test environment, e.g. when I'm running CYPRESS. Today, I could fix it by adding `revalidate: DISABLE_ISR ? 1 : <original_time>` to all of my pages using ISR, but it would be better to have a global configuration that tells all of my pages to refetch data from the server.

This would still be an issue with the new data fetching mechanism in NextJS 13, because you'd still have to do this in every `fetch` call with `revalidate` specified (and would even be worse because you could have multiple `fetch` calls with `revalidate` in a single page, as opposed to the `getStaticProps` method, which specifies one `revalidate` for the whole page)

### Describe the solution you'd like

Add a `disableISR` key to `next.config.js` that I can set using `process.env.DISABLE_ISR` (which I can set in CI).

### Describe alternatives you've considered

Can't think of any alternative solutions.

**Comment by brandonchinn178** [comment]
@balazsorban44 can you explain why this was converted into a discussion? What types of things should be an issue and what should be a discussion?

**Reply by balazsorban44** [reply]
Hi, as per our contributing guidelines, any feature request is converted to a discussion https://github.com/vercel/next.js/blob/canary/contributing/repository/triaging.md#triaging

**Reply by brandonchinn178** [reply]
Thanks! What's the next step / the best way to get Next maintainers' feedback on this?

**Reply by caioquirino** [reply]
@balazsorban44 is there any follow-up on this? Is disabling ISR something that would make sense? Thanks!

**Reply by sunlee-newyork** [reply]
+1
